x/ibc/23-commitment/README.md

```diff
@@ -0,0 +1,50 @@
+# ICS 23: Commitment
+
+Package `commitment` defines types and methods to verify other chain's state. The main type is `Store`, containing 
+proofs that can be verified when the correct value is provided. The spec functions those are directly related to 
+verification are:
+
+## Spec
+
+```typescript
+type verifyMembership = (root: CommitmentRoot, proof: CommitmentProof, key: Key, value: Value) => bool
+type verifyNonMembership = (root: CommitmentRoot, proof: CommitmentProof, key: Key) => bool
+```
+
+## Implementation
+
+### types.go
+
+`type Proof` implements `spec: type CommitmentProof`. CommitmentProof is an arbitrary object which can be used as
```

Comment:
I think generally we should declare the spec types as interfaces, so e.g. more kinds of proofs can be added later.